code/ATMOSPHERICS/atmospherics.dm

```diff
@@ -147,31 +147,33 @@ Pipelines + Other Objects -> Pipe network
 				"<span class='italics'>You hear ratchet.</span>")
 			investigate_log("was <span class='warning'>REMOVED</span> by [key_name(usr)]", "atmos")
 
-			//You unwrenched a pipe full of pressure? let's splat you into the wall silly.
+			//You unwrenched a pipe full of pressure? Let's splat you into the wall, silly.
 			if(unsafe_wrenching)
-				unsafe_pressure_release(user,internal_pressure)
+				unsafe_pressure_release(user, internal_pressure)
 			Deconstruct()
 
 	else
 		return ..()
 
 
-//Called when an atmospherics object is unwrenched while having a large pressure difference
-//with it's locs air contents.
-/obj/machinery/atmospherics/proc/unsafe_pressure_release(mob/user,pressures)
+// Throws the user when they unwrench a pipe with a major difference between the internal and environmental pressure.
+/obj/machinery/atmospherics/proc/unsafe_pressure_release(mob/user, pressures = null)
 	if(!user)
 		return
-
 	if(!pressures)
 		var/datum/gas_mixture/int_air = return_air()
 		var/datum/gas_mixture/env_air = loc.return_air()
-		pressures = int_air.return_pressure()-env_air.return_pressure()
+		pressures = int_air.return_pressure() - env_air.return_pressure()
+
+	var/dir = get_dir(src, user)
+	if(!dir)
+		dir = pick(cardinals)
+	var/turf/target = get_edge_target_turf(user, dir)
+	var/range = pressures/250
+	var/speed = range/5
 
-	var/fuck_you_dir = get_dir(src,user)
-	var/turf/general_direction = get_edge_target_turf(user,fuck_you_dir)
 	user.visible_message("<span class='danger'>[user] is sent flying by pressure!</span>","<span class='userdanger'>The pressure sends you flying!</span>")
-	//Values based on 2*ONE_ATMOS (the unsafe pressure), resulting in 20 range and 4 speed
-	user.throw_at(general_direction,pressures/10,pressures/50)
+	user.throw_at(target, range, speed)

```

Comment:
you can't ```throw_at()``` a dir, it has to be an object, hence the furthest tile on this Z in the direction of ```fuck_you_dir```